terraform/api_gateway.tf.json.template.py

```diff
@@ -93,167 +93,202 @@ def for_domain(cls, domain):
+                                        # Wildcards are not supported in `Principal`, but they are in `Condition`
+                                        "Condition": {
+                                            "StringLike": {
+                                                "aws:PrincipalArn": [f"arn:aws:iam::{account}:role/{role}"
+                                                                     for role in roles]
+                                            }
                                         }
                                     }
-                                }
-                                for account, roles in config.external_lambda_role_assumptors.items()
-                            )
-                        ]
-                    }),
-                    **aws.permissions_boundary_tf
-                }
-            },
-            "aws_iam_role_policy": {
-                lambda_.name: {
-                    "name": lambda_.name,
-                    "policy": lambda_.policy,
-                    "role": "${aws_iam_role.%s.id}" % lambda_.name
+                                    for account, roles in config.external_lambda_role_assumptors.items()
+                                )
+                            ]
+                        }),
+                        **aws.permissions_boundary_tf
+                    }
                 },
+                "aws_iam_role_policy": {
+                    lambda_.name: {
+                        "name": lambda_.name,
+                        "policy": lambda_.policy,
+                        "role": "${aws_iam_role.%s.id}" % lambda_.name
+                    },
+                }
+            } for lambda_ in lambdas
+        ),
+        {
+            "aws_wafv2_web_acl": {
+                "azul_waf_web_acl": {
+                    "name": config.qualified_resource_name("waf"),
+                    "default_action": {
+                        "allow": {}
+                    },
+                    "rule": [
+                    ],
+                    "scope": "REGIONAL",
+                    "visibility_config": {
+                        "cloudwatch_metrics_enabled": True,
+                        "metric_name": "WebACL",
+                        "sampled_requests_enabled": True,
+                    }
+                }
+            }
+        },
+        # Chalice doesn't expose the ARN of the API Gateway stages, so we
+        # construct the ARN manually using this workaround.
+        # https://github.com/aws/chalice/issues/1816#issuecomment-1012231084
+        {
+            "aws_wafv2_web_acl_association": {
+                f"azul_{lambda_}_waf": {
+                    "resource_arn": f"${{module.chalice_{lambda_}.rest_api_arn}}"
+                                    f"/stages/${{module.chalice_{lambda_}.stage_name}}",
+                    "web_acl_arn": "${aws_wafv2_web_acl.azul_waf_web_acl.arn}"
+                } for lambda_ in ('indexer', 'service')
```

Comment:
There already is a comprehension for this. Why did you roll your own? Using the existing comprehension would eliminate the need to convert it to a generator expressions.